lib/utils/fetchSDP.ts

```diff
@@ -0,0 +1,25 @@
+import { WsSdpPipeline, VideoMedia } from 'media-stream-library'
+
+export const fetchSDP = (wsURI?: string, rtspURI?: string) => {
+  const pipeline = new WsSdpPipeline({
+    ws: { uri: wsURI },
+    rtsp: { uri: rtspURI },
+  })
+
+  return pipeline.sdp.finally(() => {
+    pipeline.close()
+  })
+}
+
+export const fetchTransformationMatrix = (wsURI?: string, rtspURI?: string) => {
+  return fetchSDP(wsURI, rtspURI).then((sdp) => {
+    const videoMedia = sdp.media.find((media): media is VideoMedia => {
+      return media.type === 'video'
+    })
+    const transform = videoMedia?.transform
+    if (transform === undefined) {
+      return Promise.reject('no video media')
```

Comment:
This isn't necessarily true, it might be there is no `transform`, so either report two separate errors, or improve the error message.

Reply:
Updated.